guides/v2.2/contributor-guide/contributing_dod.md

```diff
@@ -57,74 +57,115 @@ Any backward-incompatible changes must also be recorded in the accompanying docu
+* In addition, regardless of the interpretation of the testing strategy below: Any class marked with an `@api` annotation MUST be covered with explicit test coverage via integration or unit tests. 
+  
+  These tests must test the concrete implementation's behavior in a way that can not be inadvertently changed outside of the test itself. 
+  
+  For example: testing a concrete class through DI preference is not safe because it could be overridden via configuration of another loaded module. The test should not ask for the interface but rather the concrete implementation.
+
+### Integration Tests
+
+An integration test should be used to cover any code that doesn't meet the requirements of a unit test or functional test and should be thought of as essentially the default test type.
+
+Integration tests come in many forms and can drastically vary in definition depending on what the intention of the test is. Generally speaking an integration test verifies that two or more things work together correctly. i.e. We are testing the _integration_ of one thing with another thing. 
+For the purpose of Magento testing, there are essentially two broad categories of integration tests: narrow-form and broad-form.
+
+* Narrow-form integration tests focus on testing something mostly in isolation an can be as simple as a unit test as described in this guide except without mocked dependencies.
+* Broad-form integration tests focus on testing something with less isolation and will involve multiple components explicitly collaborating. (compared to narrow-form where the dependencies are implicit collaborating)
+
+Integration test policy:
+
+* Methods and classes must have blackbox or greybox test coverage making sure to include all variations for normal use as well as corner-cases. 
+* SPI's must have test coverage. Often times Magento has extension points that may only be utilized by extension developers. Use TestModule's to implement those SPI extension points and verify they are used correctly. 
+* As mentioned above, classes marked with `@api` must contain coverage. Integration tests must be used cover these classes unless a unit test is more appropriate.
+* Consumers of default SPI's implementations should have at least basic coverage that ensures the default implementations of the SPI are correctly configured and loaded.  
+  
+  For example, `Magento\Framework\SomeClassFilter` may contain a `FilterPool` that comes with default `FilterInterface`'s from `Magento\Framework`. Each of these implementations would have their own coverage pursuant to this document. However, there should also be some basic assertions within the test coverage for `Magento\Framework\SomeClassFilter` that ensure each of the default filters are loaded correctly.     
```

Comment:
Maybe this is a bad example or could be worded better. However, I don't think the logic changes here as it scales. We still want to verify behavior of known collaborators with 10,000 filters. We would load them all anyway in the production system so if it's too slow for our test it's going to be too slow for production. 

Also the tests would be written in their respective modules. ModuleA defines the FilterPool and ModuleB defines FilterA (which of course then has a dependency on ModuleA inherently). FilterA would have isolated tests but we would have a test inside of ModuleB that loads FilterA inside of a ModuleA's FilterPool. Since this is how it would work in production, we would want to verify that it actually behaves the way we expect.